src/Background/Background.vala

```diff
@@ -242,20 +242,22 @@ namespace Gala {
             }
         }
 
-        private void load_file (string filename) {
-            if (filename.has_suffix (".xml"))
-                load_animation.begin (filename);
-            else
-                load_image (filename);
+        private inline void load_file (GLib.File file) {
+            if (file.get_basename ().has_suffix (".xml")) {
```

Comment:
i think we should use `GLib.ContentType.guess()` here, so that we can identify xml files without a extension. like the ones that the Gtk portal provides.